cachi2/core/package_managers/generic/models.py

```diff
@@ -0,0 +1,37 @@
+from typing import Literal, Optional
+
+from pydantic import BaseModel, field_validator
+
+
+class LockfileMetadata(BaseModel):
+    """Defines format of the metadata section in the lockfile."""
+
+    version: Literal["1.0"]
+
+
+class LockfileArtifact(BaseModel):
+    """Defines format of a single artifact in the lockfile."""
+
+    download_url: str
+    target: Optional[str] = None
+    checksums: dict[str, str]
+
+    @field_validator("checksums")
+    @classmethod
+    def validate_checksums(cls, value: dict[str, str]) -> dict[str, str]:
```

Comment:
nitpick: Should probably be called 'no_empty_checksums' as there isn't any validation happening here and based on the nature of the contents there can't even be any genuine validation since those strings have no semantic value.